docs/checkit.js

```diff
@@ -91,6 +88,80 @@ function CheckIt() {
 
 };
 
+CheckIt.prototype.addMonth = function() {
+    // Add a single month to the calendar
+    
+    // Get the active Calendar information because you need to know what the 
+    // last month is as well as the endDate.
+    this.store.getActive()
+        .then(function (activeCalendarId) {
+            this.store.loadById(activeCalendarId)
+                .then(function (activeCalendarState) {
+                   if (activeCalendarState !==  null) {
+                       var state = activeCalendarState;
+                       console.log(state);
+                       // Create a calendar so that you can manipulate this calendar State
+                       var calendar = new Calendar(state);
+                       // Add a month to this calendar, function has the same name
+                       // as the one we are currently in. Bad practice?
+                       calendar.addMonth();
+                       this.attachCellClickHandler(calendar, [calendar.lastMonth]);
+                   }
+               }.bind(this))
+               .catch(function(err) {
+                   console.error("Could not load calendar " + err);
+                   this.store.removeActive();
+                   this.uncollapseBuildMenu();
+               }.bind(this));
+           
+       }.bind(this))
+       
+       .catch(function () {
+           console.log("Could not load current active calendar");
+           this.uncollapseBuildMenu();
+       }.bind(this));
+    
+    
+    
+};
+
+CheckIt.prototype.attachCellClickHandler = function(calObj, monthObjArray) {
+    // Takes an array of month objects (the array can be of length 1)
+    var checkitObj = this;
+    
+    monthObjArray.forEach( function(monthObj) {
+        var $monthDiv = $('#' + monthObj.monthId);
+        $monthDiv.find('.cell').click(function(event) {
+            
+            var boxId = $(this).attr('id');
+            
+            
+            if (calObj.state.checkedDays === undefined) {
+                calObj.state.checkedDays = {};
+            }
+            //if the boxId is not checked (as in, the value is not inside of checkedDays
```

Comment:
This is somewhat difficult to follow. Maybe, "We only store data for days with checkmarks. Therefore, unchecked days are undefined."